### DISCUSSION TITLE
ARIA attributes and button disabling

### DISCUSSION THREAD

**Discussion by tobyzerner** [unpoly/unpoly#308] [Ideas]
Has any thought been given to ARIA attributes and button disabling behavior?

* In the navigation module, the `.up-current` class is applied to the current link.  Should [`[aria-current=page]`](https://www.digitala11y.com/aria-current-state/) also be applied? (Or instead, since this can be targeted with CSS!) _Addit: Ah, I see this is already implemented, just not documented!_

* Should targets currently being loaded by Unpoly be given the [`[aria-busy=true]`](https://www.digitala11y.com/aria-busy-state/) attribute?

* Should `[aria-disabled=true]` be added to a form submit button after it is clicked, and should further click events be automatically blocked? ([The Turbo folks are considering this.](https://github.com/hotwired/turbo/pull/386#issuecomment-945161995) I think it would absolutely be a sensible default.)

**Comment by thetutlage** [comment]
I think disabling the button is a good place to start with. Preventing multiple clicks is one of the reasons I want to disable buttons

**Comment by triskweline** [comment]
> In the navigation module, the .up-current class is applied to the current link. Should [aria-current=page] also be applied? (Or instead, since this can be targeted with CSS!)

Yep, we already do this. I added the missing documentation in ec81697.

> Should targets currently being loaded by Unpoly be given the [aria-busy=true] attribute?

This is implemented for the upcoming 2.5 release, in addition to a new `.up-loading` class. (2930188, 0ce1ab8).

> Should [aria-disabled=true] be added to a form submit button after it is clicked, and should further click events be automatically blocked? (The Turbo folks are considering this. I think it would absolutely be a sensible default.)

There is already an open issue waiting for someone to pick up: #104
